server/app/initializers/pubsub.rb

```diff
@@ -0,0 +1,7 @@
+if ENV['NATS_SERVERS']
+  require_relative '../services/pubsub/nats'
+  Kernel.const_set('MasterPubsub', Pubsub::Nats)
+else
+  require_relative '../services/pubsub/mongo'
+  Kernel.const_set('MasterPubsub', Pubsub::Mongo)
```

Comment:
Could the `const_set` be avoided if the classmethods in `Pubsub::Base` were moved to an actual `class MasterPubsub`, and it just varied the `Celluloid.supervise type: ...`?

I'd imagine that this kind of meta-programming magic makes the code a little hard to follow, because the `MasterPubsub` class is used everywhere, but never actually defined... feels unnecessarily complicated.

Reply:
Also, :-1: for only requiring the implementation conditionally.... the `nats` dependencies are currently going to be there anyways, and it just makes the code harder to test. There could be a syntax error in the `pubsub/nats.rb` and you wouldn't even notice unless you run the server with `NATS_SERVERS` set...